builder/comp-builder.nix

```diff
@@ -143,9 +146,38 @@ stdenv.mkDerivation ({
     # The directory containing the haddock documentation.
     # `null' if no haddock documentation was built.
     haddockDir = if doHaddock' then "${docdir drv.doc}/html" else null;
-    run = runCommand (fullName + "-run") {} ''
-      ${toString component.testWrapper} ${drv}/${installedExe} | tee $out
-    '';
+
+    # This run derivation can be used to execute test, benchmark (or even
+    # exe) components.  The $out of the derivation is a file containing
+    # the resulting stdout output.
+    run = stdenv.mkDerivation ({
+      name = (fullName + "-run");
+
+      src = drv;
```

Comment:
What if the test suite has a dependency on data files in the source directory, will it work?

Reply:
The executable is not copied out of the location it is in.  I checked that `echo $src` was the same as `echo ${drv}` in the checkPhase.  (if that is what you were concerned about).